src/Detector/Go/GoDetector.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.Common.Extensions;
+using System;
+using System.Linq;
+
+namespace Microsoft.Oryx.Detector.Go
+{
+    /// <summary>
+    /// An implementation of <see cref="IPlatformDetector"/> which detects Go applications.
+    /// </summary>
+    class GoDetector : IGoPlatformDetector
+    {
+        private readonly ILogger<GoDetector> _logger;
+        private readonly DetectorOptions _options;
+
+        /// <summary>
+        /// Creates an instance of <see cref="GoDetector"/>.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger{GoDetector}"/>.</param>
+        /// <param name="options">The <see cref="DetectorOptions"/>.</param>
+        public GoDetector(ILogger<GoDetector> logger, IOptions<DetectorOptions> options)
+        {
+            _logger = logger;
+            _options = options.Value;
+        }
+
+        public PlatformDetectorResult Detect(DetectorContext context)
+        {
+            bool isGoApp = false;
+            string appDirectory = string.Empty;
+            var sourceRepo = context.SourceRepo;
+            bool goDotModExists = false;
+            // check if go.mod exists
+            if (sourceRepo.FileExists(GoConstants.GoDotModFileName))
+            {
+                goDotModExists = true;
+                isGoApp = true;
+                _logger.LogInformation($"Found {GoConstants.GoDotModFileName} at the root of the repo. ");
+            }
+            else
+            {
+                _logger.LogDebug(
+                    $"Could not find {GoConstants.GoDotModFileName} in repo");
+                return null;
+            }
```

Comment:
I don't think these variables will be necessary if you refactor the start of this method to short-circuit if the `go.mod` file doesn't exist.

```cs
public PlatformDetectorResult Detect(DetectorContext context)
{
    if (!sourceRepo.FileExists(GoConstants.GoDotModFileName))
    {
        _logger.LogDebug(
            $"Could not find {GoConstants.GoDotModFileName} in repo");
        return null;
    }

    _logger.LogInformation($"Found {GoConstants.GoDotModFileName} at the root of the repo.");
    
    ...
}
```

But if you're adding more checks for what specifies a Go repository, than it would make sense to these variables (or some separate context object) to track what files were found that determined the repository was a Go repository.

Reply:
👍  there are a couple additional checks in the TODO that are being considered